xclim/indices/_threshold.py

```diff
@@ -382,13 +383,16 @@ def snd_season_end(
     thresh = convert_units_to(thresh, snd)
     cond = snd >= thresh
 
-    out = (
+    resampled = (
         cond.resample(time=freq)
         .map(rl.season, window=window, dim="time", coord="dayofyear")
         .end
     )
-    out.attrs.update(units="", is_dayofyear=np.int32(1), calendar=get_calendar(snd))
-    return out.where(~valid)
+    resampled.attrs.update(
+        units="", is_dayofyear=np.int32(1), calendar=get_calendar(snd)
+    )
+    out: xarray.DataArray = resampled.where(~valid)
```

Comment:
Same as above for `sel` : wouldn't xarray's typing make this clear ?

Reply:
This is an issue with the signature for `convert_units_to`; If we want to have consistency between what is passed and what is received, we need to be using Generics.

However, we can't use `Quantified` because the output is either a `DataArray` or a `float`. It's not possible for the checker to determine the combination of call signatures needed to get either of these, so the end result is that the received variable is potentially both types. Therefore, we need to constrain it to `DataArray` on reception.

Reply:
The documentation on this is a bit technical. Maybe we can tackle this in another PR?

See: https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators

Reply:
Honestly, I think it would be better to wait and fix these issues instead of introducing new things that we know will be removed. Or at least, this PR shouldn't add annotations that we know will be removed by the subsequent PR ?

Mypy tells you when types are not correctly used or are unparsable, but it does not tell you about unnecessary annotations, does it ?

Reply:
It does not warn us about redundant annotations, no. Should I go forward with the removal of these specific changes, do you think?